pipeline/gpt.py

```diff
@@ -4,8 +4,7 @@
 import dspy.utils
 
 MODELS_WITH_RESPONSE_FORMAT = [
```

Comment:
We should properly handle the case where the deployment_id passed to GPT is not in MODELS_WITH_RESPONSE_FORMAT?